src/shared/mod.rs

```diff
@@ -94,11 +93,57 @@ pub fn apply_changes(
     original
 }
 
+pub fn create_ast_package(
+    uri: String,
+) -> Result<flux::ast::Package, String> {
+    let values = cache::get_package(uri.clone())?;
+
+    let pkgs = values
+        .into_iter()
+        .map(|v: cache::CacheValue| {
+            utils::create_file_node_from_text(
+                v.uri.clone(),
+                v.contents,
+            )
+        })
+        .collect::<Vec<flux::ast::Package>>();
+
+    let pkg = pkgs.into_iter().fold(
+        None,
+        |acc: Option<flux::ast::Package>, pkg| {
```

Comment:
I'm curious, why is the `Option` needed here?

Reply:
because we need to start at a state of None as its possible that get_package will return no values, we build off the first pkg after it is confimed there is a first one.

Reply:
ahh ok, thanks 👍